benchmark.py

```diff
@@ -58,36 +67,43 @@ def main():
         "MAX_INPUT_TOKEN_LENGTH": MAX_INPUT_TOKEN_LENGTH,
     }
 
-    tic = time.perf_counter()
+    # Initialization
+    init_tic = time.perf_counter()
     llama2_wrapper = LLAMA2_WRAPPER(config)
     llama2_wrapper.init_tokenizer()
     llama2_wrapper.init_model()
-    toc = time.perf_counter()
-    print(f"Initialize the model in {toc - tic:0.4f} seconds.")
-
-    example = "Can you explain briefly to me what is the Python programming language?"
-
-    generator = llama2_wrapper.run(
-        example, [], DEFAULT_SYSTEM_PROMPT, DEFAULT_MAX_NEW_TOKENS, 1, 0.95, 50
-    )
-    tic = time.perf_counter()
-    try:
-        first_response = next(generator)
-        # history += [(example, first_response)]
-    except StopIteration:
-        pass
-        # history += [(example, "")]
-    for response in generator:
-        # history += [(example, response)]
-        pass
-    print(response)
-
-    toc = time.perf_counter()
-    output_token_length = llama2_wrapper.get_token_length(response)
-
-    print(f"Generating the out in {toc - tic:0.4f} seconds.")
-    print(f"Speed: {output_token_length / (toc - tic):0.4f} tokens/sec.")
-
+    init_toc = time.perf_counter()
+    initialization_time = init_toc - init_tic
+
+    total_time = 0
+    total_tokens_per_second = 0
+    total_memory_gen = 0
+
+    prompt_example = "Can you explain briefly to me what is the Python programming language?"
+    model_response = ""
+
+    # Cold run
+    print("Performing cold run...")
+    run_iteration(llama2_wrapper, prompt_example, DEFAULT_SYSTEM_PROMPT, DEFAULT_MAX_NEW_TOKENS)
+
+    # Timed runs
+    print("Performing timed runs...")
```

Comment:
Consider adding  `iter` number in the printout like ` print(f"Performing {args.iter} timed runs...")` .